src/cdi_ndi/NDI_Base.cc

```diff
@@ -23,26 +23,33 @@ namespace rtt_cdi_ndi {
  * For more details on NDI, see
  * https://xweb.lanl.gov/projects/data/nuclear/ndi/ndi.html
  *
- * \param[in] gendir_in path to gendir file
  * \param[in] dataset_in name of requested dataset (provided by inherited class)
  * \param[in] library_in name of requested NDI data library
  * \param[in] reaction_in name of requested reaction
- * \param[in] mg_form_in name of multigroup energy discretization to use
+ * \param[in] mg_e_bounds_in multigroup energy bin boundaries (keV)
  */
-NDI_Base::NDI_Base(const std::string &gendir_in, const std::string &dataset_in,
-                   const std::string &library_in,
-                   const std::string &reaction_in, const MG_FORM mg_form_in)
-    : gendir(gendir_in), dataset(dataset_in), library(library_in),
-      reaction(reaction_in), mg_form(mg_form_in) {
+NDI_Base::NDI_Base(const std::string &dataset_in, const std::string &library_in,
+                   const std::string &reaction_in,
+                   const std::vector<double> mg_e_bounds_in)
+    : dataset(dataset_in), library(library_in), reaction(reaction_in),
+      mg_e_bounds(mg_e_bounds_in) {
+  gendir = std::string(NDI_ROOT_DIR) + "share/gendir.all";
 
   Require(gendir.length() > 0);
   Require(dataset.length() > 0);
   Require(library.length() > 0);
   Require(reaction.length() > 0);
+  Require(mg_e_bounds.size() > 0);
 
-  mg_form_map.insert(mg_pair(MG_FORM::LANL4, "4_lanl"));
-  mg_form_map.insert(
-      mg_pair(MG_FORM::NOT_SET, "No multigroup representation set!"));
+  for (size_t i = 0; i < mg_e_bounds.size(); i++) {
+    mg_e_bounds[i] /= 1000.; // keV -> MeV
+  }
+
+  // Check that mg_e_bounds is monotonically decreasing (NDI requirement)
```

Comment:
You might be able to use a function from Draco's ds++/dbc.hh to do this! (There's an "is_strict_monotonic_decreasing" function in there, I think!)

Reply:
whoa nice -- thanks!